core/src/test/java/com/datastax/oss/driver/internal/core/channel/InFlightHandlerTest.java

```diff
@@ -311,6 +313,65 @@ public void should_close_gracefully_if_orphan_ids_above_max_and_pending_requests
     assertThat(channel.closeFuture()).isSuccess();
   }
 
+  @Test
+  public void should_close_gracefully_if_orphan_ids_above_max_and_multiple_pending_requests() {
```

Comment:
`should_close_gracefully_if_orphan_ids_above_max_and_pending_request` feels a bit redundant now with this test case, but I think that's probably alright and there's some value in testing to 1 past the threshold / multiple past the threshold.